03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,421 @@
+
+```{r}
+# Calculate the variance for each gene
+variances <- apply(assay(dds_norm), 1, var)
+
+# Determine the upper quartile variance cutoff value
+upper_var <- quantile(variances, 0.75)
+
+# Subset the data choosing only genes whose variances are in the upper quartile
+df_by_var <- data.frame(assay(dds_norm)) %>%
+  dplyr::filter(variances > upper_var)
+```
+
+## Create a heatmap
+
+To further customize the heatmap, see a vignette for a guide at this [link](https://slowkow.com/notes/pheatmap-tutorial/) [@make-heatmaps].
+
+```{r}
+# Create and store the heatmap object
+pheatmap <-
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE, # We want to cluster the heatmap by rows (genes in this case)
+    cluster_cols = TRUE, # We also want to cluster the heatmap by columns (samples in this case),
+    show_rownames = FALSE, # We don't want to show the rownames because there are too many genes for the labels to be clearly seen
+    main = "Non-Annotated Heatmap",
+    colorRampPalette(c(
+      "deepskyblue",
+      "black",
+      "yellow"
+    ))(25
+    ),
+    scale = "row" # Scale values in the direction of genes (rows)
+  )
+```
+
+We've created a heatmap but although our genes and samples are clustered, there
+is not much information that we can gather here because we did not provide the
+`pheatmap` function with annotation labels for our samples.
+
+First let's save our clustered heatmap.
+
+### Save heatmap as a png
+
+You can easily switch this to save to a jpeg or tiff by changing the function 
+and file name within the function to the respective file suffix.
+
+```{r}
+# Open a png file
+png(file.path(
+  plots_dir,
+  "HeatmapSRP070849_nonannotated.png" # Replace file name with a relevant output plot name
+))
+
+# Print your heatmap
+pheatmap
+
+# Close the png file:
+dev.off()
+```
+
+Now, let's add some annotation bars to our heatmap.
+
+## Prepare metadata for annotation
+
+We are going to manipulate the metadata and add variables with the information that we would like to use to annotate the heatmap.
+
+```{r}
+# Let's prepare the annotation data.frame for the uncollapsed `DESeqData` set object which will be used to create the technical replicates heatmap
+annotation_df <- metadata %>%
+  # Create a variable to store the cancer type information
+  dplyr::mutate(
+    mutation = dplyr::case_when(
+      startsWith(refinebio_title, "TET2") ~ "TET2",
+      startsWith(refinebio_title, "IDH2") ~ "IDH2",
+      startsWith(refinebio_title, "WT") ~ "WT",
+      TRUE ~ "unknown"
+    )
+  ) %>%
+  # We want to select the variables that we want for annotating the technical replicates heatmap
+  dplyr::select(
+    refinebio_accession_code,
+    mutation,
+    refinebio_treatment
+  ) %>%
+  # The `pheatmap` function requires that the rownames of our annotation object matches the column names of our `DESeaDataSet` object
```

Comment:
```suggestion
  # The `pheatmap()` function requires that the row names of our annotation object matches the column names of our `DESeaDataSet` object
```

Reply:
I will add a tidbit in the style guide about when we use row names vs rownames()